🐛 Bug: eslintignore should include coverage directory

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

ESLint shouldn't attempt to lint `.js` files in `coverage/` after `pnpm run test --coverage` is run.

### Actual

It does.

```plaintext
/Users/josh/repos/template-typescript-node-package/coverage/lcov-report/block-navigation.js
  1:0  error  Requires 'eslint-enable' directive                                                 eslint-comments/disable-enable-pair
  1:0  error  Unexpected unlimited 'eslint-disable' comment. Specify some rule names to disable  eslint-comments/no-unlimited-disable
```

### Additional Info

_No response_

--- a/.eslintignore
+++ b/.eslintignore
@@ -1,1 +1,2 @@
+coverage
 lib

